Index: inc/core/NyARDoubleMatrix33.h
===================================================================
--- inc/core/NyARDoubleMatrix33.h	(revision 259)
+++ inc/core/NyARDoubleMatrix33.h	(working copy)
@@ -45,7 +45,7 @@
 		double m21;
 		double m22;
 	public:
-		static NyArray<NyARDoubleMatrix33>* NyARDoubleMatrix33::createArray(int i_number);
+		static NyArray<NyARDoubleMatrix33>* createArray(int i_number);
 	public:
 		void setValue(const double i_value[]);
 		void getValue(double o_value[])const;
Index: inc/core/NyARLabelingLabelStack.h
===================================================================
--- inc/core/NyARLabelingLabelStack.h	(revision 259)
+++ inc/core/NyARLabelingLabelStack.h	(working copy)
@@ -34,8 +34,8 @@
 	class NyARLabelingLabelStack :public NyObjectStack<NyARLabelingLabel>
 	{
 	public:
-		NyARLabelingLabelStack::NyARLabelingLabelStack(int i_length);
-		virtual NyARLabelingLabelStack::~NyARLabelingLabelStack(void);
+		NyARLabelingLabelStack(int i_length);
+		virtual ~NyARLabelingLabelStack(void);
 	protected:
 		void onReservRequest(int i_start, int i_end, NyArray<NyARLabelingLabel*>& i_buffer);
 	public:
Index: inc/core/NyARSquareDetector.h
===================================================================
--- inc/core/NyARSquareDetector.h	(revision 259)
+++ inc/core/NyARSquareDetector.h	(working copy)
@@ -66,7 +66,7 @@
 		*/
 	public:
 		NyARSquareDetector(const NyARCameraDistortionFactor& i_dist_factor_ref,const TNyARIntSize& i_size);
-		NyARSquareDetector::~NyARSquareDetector();
+		~NyARSquareDetector();
 	private:
 		int _max_coord;
 
